lib/puppet/provider/cisco_vxlan_global/nxapi.rb

```diff
@@ -46,7 +46,7 @@
 
   def initialize(value={})
     super(value)
-    @vxlan_global = Cisco::VxlanGlobal.new
+    @vxlan_global = Cisco::VxlanGlobal.new if Cisco::VxlanGlobal.enabled
```

Comment:
@mikewiebe We worked through this during consult hours. The problem is that the NU code will instantiate (by default) when you call VxlanGlobal.new; so self.instances gets a new instance every time, meaning ensure => :present is true; meaning destroy gets called every time, breaking idempotence.